lib/web-server.js

```diff
@@ -93,6 +93,8 @@ function createWebServer (injector, config) {
 
   server.on('upgrade', function (req, socket, head) {
     log.debug(`upgrade ${req.url}`)
+    // ignore future socket errors
+    socket.on('error', () => {})
```

Comment:
Seems like this will cause all errors to be ignored after we start the two-way socket-io.

The spurious errors from #3295 only occur after the run completes.   Isn't the time to turn off errors?

Reply:
I would say that you don't want those socket errors to crash the entire Karma process and this prevents just that. It still allows for the upgrade handler(s) to attach their own error event handlers.

Reply:
In fact. I think this is somewhat similar to what Node.js does when it's HTTP server receives a new connection. It attaches an error handler to prevent it from going down. It forwards it to the 'clientError' event on the server, which when emitted does not crash if it does not have any handlers. But in the case if an upgrade request it totally lets go of the socket and removes that error handler so the socket is entirely yours.

Reply:
Sorry I missed your reply.

Yes, just change the comment.

Reply:
> The spurious errors from #3295 only occur after the run completes. Isn't the time to turn off errors?

I agree. While this fixes the problem it will swallow errors in other situations. At the bare minimum, it should log an error as a warning similar to https://github.com/karma-runner/karma/blob/master/lib/server.js#L145. Ideally, it should also reset the handler after the complete event is received to target specifically https://github.com/karma-runner/karma/issues/3295, where we don't care about the browser errors anymore.